### PR TITLE
Bugfix/762 fixprevnext

### PR DESCRIPTION
**Pull request by mrblippy**
#762

### CODE REVIEW

**Comment by edalex-ian** [comment]
So just for context @mrblippy , you did incremental testing of this branch while @doolse was working on it right?

As a result we're confident it fixes the issue? (But obviously we'll need to test a good number of other areas.

Also, we should discuss the back-portability of this - as there was a need for it in 2018.2. :thinking:

File: Source/Plugins/Core/com.equella.core/src/com/tle/mycontent/web/selection/MyContentSelectSection.java [hunk 1]
```diff
@@ -36,4 +36,9 @@ protected String getSessionKey() {
   public Label getTitle(SectionInfo info) {
     return LABEL_TITLE;
   }
+
+  @Override
+  protected String getPageName() {
+    return null;
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/src/com/tle/mycontent/web/selection/MyContentSelectSection.java]: What's going on here? Why are all these pagenames set to null?

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/itemlist/item/AbstractItemListEntry.java [hunk 2]
```diff
@@ -80,16 +83,23 @@
...
   @Inject private HtmlEditorService htmlEditorService;
   @Inject private ItemService itemService;
+  private List<BookmarkModifier> itemUrlModifiers = new ArrayList<>();
 
   @Override
   protected Bookmark getTitleLink() {
     final Item item = getItem();
     if (item != null) {
-      return itemUrls.createItemUrl(info, item.getItemId());
+      ViewItemUrl vurl = itemUrls.createItemUrl(info, item.getItemId());
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/itemlist/item/AbstractItemListEntry.java]: should probably be vUrl or VIUrl

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/itemlist/item/StandardItemListEntryFactory.java [hunk 3]
```diff
@@ -55,6 +57,10 @@ public StandardItemListEntry createItemListItem(
     if (itemListItem == null) {
       itemListItem = entryFactory.get();
     }
+    String spage = info.getTreeAttribute(ContextableSearchSection.SEARCHPAGE_ATTR);
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/itemlist/item/StandardItemListEntryFactory.java]: should probably be sPage, or searchPage.

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/searching/prevnext/SearchPrevNextSection.java [hunk 4]
```diff
@@ -78,142 +72,34 @@ public void registered(String id, SectionTree tree) {
...
-
-    // if we've passed beyond the limited paged result set so far mapped, we
-    // may need to manually insert the new ItemId into the session map
-    ItemId keyAsItemId = new ItemId(itemKeyToView.getUuid(), itemKeyToView.getVersion());
-    model.setCurrentItemKey(keyAsItemId);
-    if (model.getIndexIntoSearch() < 0) {
-      model.getMappedSearchIndexValues().mapItemIdWithIndex(keyAsItemId, newIndex, false);
+    int nextIndex = model.getIndex() + direction;
+    SectionInfo searchInfo = info.createForward(model.getSearchPage());
+    AbstractFreetextResultsSection srs =
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/searching/prevnext/SearchPrevNextSection.java]: srs and ftr should probably be renamed. Abbreviating like this adds ambiguity to the code.

**Pull request merged**